### PR TITLE
DAOS-2240 Add Jenkins RPM project repos to provisioned nodes

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

**Comment by brianjmurrell** [comment]
[Tested](https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-104/2/) with daos-stack/cart#104 to make sure non-RPM builds still do the right thing and passes.

File: vars/provisionNodes.groovy [hunk 1]
```diff
@@ -113,14 +113,22 @@ def call(Map config = [:]) {
                               chmod 600 /localhome/jenkins/.ssh/{authorized_keys,id_rsa*}
                               chown -R jenkins.jenkins /localhome/jenkins/.ssh
                               echo \\"jenkins ALL=(ALL) NOPASSWD: ALL\\" > /etc/sudoers.d/jenkins
+                              yum -y install yum-utils
+                              pkgs=\\"openpa libfabric mercury pmix ompi\\"
+                              for ext in \\$pkgs; do
+                                  rm -f /etc/yum.repos.d/build.hpdd.intel.com_job_daos-stack_job_\\"\\${ext}\\"_job_*.repo
+                                  yum-config-manager --add-repo=http://build.hpdd.intel.com/job/daos-stack/job/\\"\\${ext}\\"/job/master/lastSuccessfulBuild/artifact/artifacts/
+                                  echo \\"gpgcheck = False\\" >> /etc/yum.repos.d/build.hpdd.intel.com_job_daos-stack_job_\\"\\${ext}\\"_job_master_lastSuccessfulBuild_artifact_artifacts_.repo
```
Comment (JohnMalmberg) [inline vars/provisionNodes.groovy]: Should not hard code in path here.  This needs to work with build-stage/build-dev, etc.

File: vars/provisionNodes.groovy [hunk 2]
```diff
@@ -113,14 +113,22 @@ def call(Map config = [:]) {
...
+                              done
```
Comment (JohnMalmberg) [inline vars/provisionNodes.groovy]: Should not hard-code the URLs here.  This needs to work for build-stage, build-dev, etc.

Comment (brianjmurrell) [reply vars/provisionNodes.groovy]: Right.  Fixed in latest commit.

**Pull request merged**